lib/routes/pconline/focus.ts

```diff
@@ -0,0 +1,152 @@
+import got from '@/utils/got';
+import { load } from 'cheerio';
+import { parseDate } from '@/utils/parse-date';
+
+const rootUrl = 'https://www.pconline.com.cn';
+interface Item {
+    id: string;
+    title: string;
+    channelName: string;
+    wap_3g_url: string;
+    bigImage: string;
+    cover: string;
+    largeImage: string;
+    authorname: string;
+    authorImg: string;
+    pc_pubDate: string;
+    artType: string;
+    summary: string;
+    url: string;
+}
+
+const categories = {
+    all: {
+        title: '全部',
+        path: '',
+    },
+    tech: {
+        title: '科技',
+        path: 'tech/',
+    },
+    finance: {
+        title: '财经',
+        path: 'finance/',
+    },
+    life: {
+        title: '生活',
+        path: 'life/',
+    },
+    company: {
+        title: '公司',
+        path: 'company/',
+    },
+    character: {
+        title: '人物',
+        path: 'character/',
+    },
+};
+
+const getContent = (item) =>
+    cache.tryGet(item.link, async () => {
+        const detailResponse = await got({
+            method: 'get',
+            url: `https:${item.link}`,
+            responseType: 'arrayBuffer',
+        });
+
+        const utf8decoder = new TextDecoder('GBK');
+        const html = utf8decoder.decode(detailResponse.data);
+        const $ = load(html);
+        item.description = $('.context-box .context-table tbody td').html();
+        return item;
+    });
+
+export const handler = async (ctx) => {
+    const { category = 'all' } = ctx.req.param();
+    const cate = categories[category] || categories.all;
+    const currentUrl = `${rootUrl}/3g/other/focus/${cate.path}index.html`;
+
+    let items = [];
+
+    try {
+        let dataString = await cache.get(currentUrl);
+        if (!dataString) {
+            const response = await got({
+                method: 'get',
+                url: currentUrl,
+            });
+
+            const resString = response.data
+                .replace(/Module\.callback\((.*)\)/s, '$1')
+                .split('\n')
+                .filter((e) => e.indexOf('"tags":') !== 0)
+                .join('\n')
+                .replaceAll("'", '"');
+            const tinyData = resString.replaceAll(/[\n\r]/g, '');
+            dataString = tinyData.replaceAll(',}', '}');
+        }
+        const data = JSON.parse(dataString || '');
+        const { articleList } = data;
+        items = articleList.map((item: Item) => ({
+            id: item.id,
+            title: item.title,
+            author: [
+                {
+                    name: item.authorname,
+                    avatar: item.authorImg,
+                },
+            ],
+            pubDate: parseDate(item.pc_pubDate),
+            link: item.url,
+            description: item.summary,
+            category: item.channelName,
+            image: item.cover,
+        }));
+        await Promise.all(items.map((item) => getContent(item)));
+        cache.set(currentUrl, JSON.stringify(items));
```

Comment:
https://docs.rsshub.app/joinus/advanced/use-cache#cache-tryget-key-getvaluefunc-maxage-refresh

> You should use `cache.tryGet()` most of the time.

Reply:
OK，Thanks you